### DISCUSSION TITLE
Card的onHeaderClick不能调组件吗

### DISCUSSION THREAD

**Discussion by githubqh** [ant-design/ant-design-mobile#5240] [Q&A] [answered]
### Description

https://codesandbox.io/s/focused-keller-nntjzf?file=/src/App.tsx
![image](https://user-images.githubusercontent.com/105548468/170432485-f9f477fe-afb1-4819-a721-ae6ac0ca6e35.png)

**Comment by awmleer** [comment]
看起来是一个组件用法上的疑问，我来把它转移到 discussion 板块

**Reply by githubqh** [reply]
好的

**Comment by GOWxx** [comment, marked as answer]
```typescript
type CardProps = {
    onHeaderClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
}
```
可以看到 `onHeaderClick` 方法里 `return` 并不会有什么作用
如果是想在本页面渲染浮动页面或组件，可以看看 `Modal` `Popup` 或者别的符合需求的组件
如果没有符合需求的话，可以考虑自己写个组件，`onHeaderClick` 方法里改变变量控制该组件的显示隐藏状态

在你的 `demo` 上加了 `Popup`
https://codesandbox.io/s/happy-thompson-fk7hdk?file=/src/App.tsx

**Reply by githubqh** [reply]
明白了，感谢大神

**Reply by githubqh** [reply]
> ```ts
> type CardProps = {
>     onHeaderClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
> }
> ```
> 
> 可以看到 `onHeaderClick` 方法里 `return` 并不会有什么作用 如果是想在本页面渲染浮动页面或组件，可以看看 `Modal` `Popup` 或者别的符合需求的组件 如果没有符合需求的话，可以考虑自己写个组件，`onHeaderClick` 方法里改变变量控制该组件的显示隐藏状态
> 
> 在你的 `demo` 上加了 `Popup` https://codesandbox.io/s/happy-thompson-fk7hdk?file=/src/App.tsx

明白了！！非常感谢
